### ISSUE TITLE
404 when publishing an extension

### ISSUE THREAD

**Issue by waderyan** [Microsoft/vscode#3424]
Copied from internal email. 

I'm trying to publish a Visual Studio Code extension. I followed the [steps in the docs](https://code.visualstudio.com/docs/tools/vscecli) but, after running vsce publish, I get the following error:

`Error: Failed Request: Bad Request(400) - TF400898: An Internal Error Occurred. Activity Id: 587a31af-2cd4-40ae-bed1-acd5c03723a4.`

**Comment by rid9** [comment]
This is what I did:

1. Went to my account's home page, then to My Profile.

2. Under Security, I created a new Personal Access Token.

3. Used "vsce" as the description, set the expiration to 1 year, selected "All accessible accounts" from the Accounts dropdown, kept Authorized Scopes to "All scopes", then clicked "Create Token".

4. Used `vsce create-publisher rid9`:

> ```
> Publisher human-friendly name: (rid9) rid9
> Personal Access Token: ****************************************************
> 
> Successfully created publisher 'rid9'.
> ```
> 

5. Used `vsce login rid9`:

> ```
> Personal Access Token for publisher 'rid9': ****************************************************
> 
> Authentication successful. Found publisher 'rid9'.
> ```
> 

6. Used `vsce publish`:

> ```
> Executing prepublish script 'node ./node_modules/vscode/bin/compile'...
> Publishing rid9.datetime@1.0.0...
> Error: Failed Request: Bad Request(400) - TF400898: An Internal Error Occurred. Activity Id: 50ad89d2-fd2c-48a2-840c-cda0e56cc832.
> ```

---

These are the requests that `vsce publish` made:

1. An `OPTIONS` request to https://marketplace.visualstudio.com/_apis/gallery which returned a 200 response containing JSON data.

2. A `GET` request to https://marketplace.visualstudio.com/_apis/gallery/publishers/rid9/extensions/datetime?flags=1 which returned a 404 response containing the following JSON data:

> ```json
> {
> 	"$id": "1",
> 	"innerException": null,
> 	"message": "The requested extension doesn't exist.",
> 	"typeName": "Microsoft.VisualStudio.Services.Gallery.WebApi.ExtensionDoesNotExistException, Microsoft.VisualStudio.Services.Gallery.WebApi, Version=14.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a",
> 	"typeKey": "ExtensionDoesNotExistException",
> 	"errorCode": 0,
> 	"eventId": 3000
> }
> ```

3. A `POST` request to https://marketplace.visualstudio.com/_apis/gallery/extensions, with a `Content-Length` of `828509` and a JSON body containing `{ "extensionManifest": "(base64 encoded string)" }`, which returned a 400 response containing the following JSON data:

> ```json
> {
> 	"$id": "1",
> 	"innerException": null,
> 	"message": "TF400898: An Internal Error Occurred. Activity Id: 50ad89d2-fd2c-48a2-840c-cda0e56cc832.",
> 	"typeName": "System.InvalidOperationException, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089",
> 	"typeKey": "InvalidOperationException",
> 	"errorCode": 0,
> 	"eventId": 0
> }
> ```

**Comment by waderyan** [comment]
@joaomoreno can you take a look at this?

**Comment by joaomoreno** [comment]
This issue was moved to Microsoft/vscode-vsce#68

**Comment by joaomoreno** [comment]
@waderyan  `Internal Error Occurred` means it's a service problem, we should inform them.

Moving to `vsce`. Created bug in Microsoft/vso-node-api#36 to see if we can get their attention.

**Comment by joaomoreno** [comment]
@rid9 Can you publish any other extensions?

**Comment by rid9** [comment]
@joaomoreno I found what the problem was. The name of the extension was "Date & Time" and the description contained "Date & Time" as well. When I renamed it to "Date and Time" and removed all references to `&` from the name and description, I succeeded in publishing it.

I also noticed that the generated package contains an `extension.vsixmanifest` XML file with a `<DisplayName/>` and a `<Description/>` tag matching the values from `package.json`. So I unpublished it, replaced all `&`s with `&amp;` in this file, then succeeded in publishing the extension again, this time with the name and description containing `&`.

I created this pull request: https://github.com/Microsoft/vscode-vsce/pull/69
